### PR TITLE
Avoid the need to bring up audio engine in test environment

### PR DESCRIPTION
**Pull request by notspiff**
I would have liked to put the checking function in the service broker but only a reference is exposed so it has to be lifted to manager.

### CODE REVIEW

File: xbmc/ServiceManager.h [hunk 1]
```diff
@@ -113,6 +113,7 @@ class CServiceManager
   XBPython& GetXBPython();
 #endif
   PVR::CPVRManager& GetPVRManager();
+  bool HasActiveAE() const;
```
Comment (xhaggi) [inline xbmc/ServiceManager.h]: Should we add a delegate to our service broker so we are able to drop the use of `g_application.m_ServiceManager`?

Comment (notspiff) [reply xbmc/ServiceManager.h]: WTF was unclear about my pr description?

**Comment by FernetMenta** [comment]
imo good for now. the more we can separate optional components, the more we should think about a better method then ServiceBroker returning references.

File: xbmc/ServiceManager.h (hunk 1 above)
Comment (notspiff) [reply xbmc/ServiceManager.h]: and that is not straight forward as shit use the service manager directly elsewhere so then i have to change those spots from . to ->.

Comment (xhaggi) [reply xbmc/ServiceManager.h]: Ok sorry for the noise

Comment (notspiff) [reply xbmc/ServiceManager.h]: nw. i spewed some frustrations out on you which was not fair, sorry for that.

Comment (xhaggi) [reply xbmc/ServiceManager.h]: Np .. all good.

**Comment by notspiff** [comment]
no hard objections, and this brings tests back in working order. it's intended as an incremental improvement and not the cure for world hunger -> merging.

**Pull request merged**